shell/platform/linux/fl_platform_views_plugin.cc

```diff
@@ -0,0 +1,310 @@
+
+  const gchar* method = fl_method_call_get_name(method_call);
+  FlValue* args = fl_method_call_get_args(method_call);
+
+  g_autoptr(FlMethodResponse) response = nullptr;
+  if (strcmp(method, kCreateMethod) == 0) {
+    response = platform_views_create(self, args);
+  } else if (strcmp(method, kDisposeMethod) == 0) {
+    response = platform_views_dispose(self, args);
+  } else if (strcmp(method, kAcceptGestureMethod) == 0) {
+    response = platform_views_accept_gesture(self, args);
+  } else if (strcmp(method, kRejectGestureMethod) == 0) {
+    response = platform_views_reject_gesture(self, args);
+  } else if (strcmp(method, kEnterMethod) == 0) {
+    response = platform_views_enter(self, args);
+  } else if (strcmp(method, kExitMethod) == 0) {
+    response = platform_views_exit(self, args);
+  } else {
+    response = FL_METHOD_RESPONSE(fl_method_not_implemented_response_new());
+  }
+
+  if (response != nullptr) {
+    send_response(method_call, response);
+  }
+}
+
+static void fl_platform_views_plugin_dispose(GObject* object) {
+  FlPlatformViewsPlugin* self = FL_PLATFORM_VIEWS_PLUGIN(object);
+
+  g_clear_object(&self->channel);
+  g_hash_table_destroy(self->factories);
+  g_hash_table_destroy(self->platform_views);
+  self->factories = self->platform_views = nullptr;
+
+  G_OBJECT_CLASS(fl_platform_views_plugin_parent_class)->dispose(object);
+}
+
+static void fl_platform_views_plugin_class_init(
+    FlPlatformViewsPluginClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_platform_views_plugin_dispose;
+}
+
+static void fl_platform_views_plugin_init(FlPlatformViewsPlugin* self) {}
+
+FlPlatformViewsPlugin* fl_platform_views_plugin_new(
+    FlBinaryMessenger* messenger,
+    FlGestureHelper* gesture_helper) {
+  g_return_val_if_fail(FL_IS_BINARY_MESSENGER(messenger), nullptr);
+
+  FlPlatformViewsPlugin* self = FL_PLATFORM_VIEWS_PLUGIN(
+      g_object_new(fl_platform_views_plugin_get_type(), nullptr));
+
+  g_autoptr(FlStandardMethodCodec) codec = fl_standard_method_codec_new();
+  self->channel =
+      fl_method_channel_new(messenger, kChannelName, FL_METHOD_CODEC(codec));
+  fl_method_channel_set_method_call_handler(self->channel, method_call_cb, self,
+                                            nullptr);
+
+  self->gesture_helper = gesture_helper;
+  self->factories = g_hash_table_new(g_str_hash, g_str_equal);
+  self->platform_views = g_hash_table_new(g_direct_hash, g_direct_equal);
+
+  return self;
+}
+
+gboolean fl_platform_views_plugin_register_view_factory(
+    FlPlatformViewsPlugin* self,
+    FlPlatformViewFactory* factory,
+    const gchar* view_type) {
+  return g_hash_table_insert(self->factories, g_strdup(view_type), factory);
+}
+
+FlPlatformView* fl_platform_views_plugin_get_platform_view(
+    FlPlatformViewsPlugin* self,
+    int view_identifier) {
+  return (FlPlatformView*)g_hash_table_lookup(self->platform_views,
+                                              GINT_TO_POINTER(view_identifier));
```

Comment:
Can this be a safe `FL_PLATFORM_VIEW()` cast?

Reply:
It's safe, because we have checked its type in `platform_views_create` before inserting it into `platform_views`.